### PR TITLE
More achievements!

### PR DESCRIPTION
**Pull request by Jaaslet**
More achievements!
- Sevii is getting big. The plan is to move all sevii stuff to it's own category and give it a total of 50%. Also adding achievements for sevii gyms.
- I want to add a bunch of new global achievements. I am bumping the global bonus to 200%, so it does not get watered down too much.

### CODE REVIEW

**Comment by Jaaslet** [comment]
I don't really have a nice way of moving the Sevii dungeon achievements. But i have marked it as ready for review, in case we want to merge it, before i come up with a better solution.

File: src/modules/requirements/QuestLevelRequirement.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    constructor(levelRequired: number) {
+        super(levelRequired, GameConstants.AchievementOption.more, GameConstants.AchievementType.Quest);
+    }
+
+    public getProgress() {
+        return Math.min(App.game.quests.level(), this.requiredValue);
+    }
+
+    public hint(): string {
+        return `Needs questlevel ${this.requiredValue}.`;
```
Comment (Aegyo) [inline src/modules/requirements/QuestLevelRequirement.ts:14]:
```suggestion
        return `Needs quest level ${this.requiredValue}.`;
```

File: src/scripts/achievements/AchievementHandler.ts [hunk 2]
```diff
@@ -177,13 +194,21 @@ class AchievementHandler {
...
         AchievementHandler.addAchievement('Token Collector', 'Obtain 1,000 Dungeon Tokens', new TokenRequirement(1000), 0.10);
         AchievementHandler.addAchievement('Dungeon Grinder', 'Obtain 10,000 Dungeon Tokens', new TokenRequirement(10000), 0.15);
         AchievementHandler.addAchievement('Dungeon Master', 'Obtain 100,000 Dungeon Tokens', new TokenRequirement(100000), 0.20);
         AchievementHandler.addAchievement('Dungeon Legend', 'Obtain 1,000,000 Dungeon Tokens', new TokenRequirement(1000000), 0.25);
+        AchievementHandler.addAchievement('Dungeon Myth', 'Obtain 100,000,000 Dungeon Tokens', new TokenRequirement(100000000), 0.3);
+        AchievementHandler.addAchievement('Dungeon God', 'Obtain 10,000,000,000 Dungeon Tokens', new TokenRequirement(10000000000), 0.6);
+
+        AchievementHandler.addAchievement('Farm Master', 'Obtain 100,000 Farm Points', new FarmPointsRequirement(100000), 0.20);
+        AchievementHandler.addAchievement('Farm Legend', 'Obtain 1,000,000 Farm Points', new FarmPointsRequirement(1000000), 0.30);
+        AchievementHandler.addAchievement('Farm Myth', 'Obtain 20,000,000 Farm Points', new FarmPointsRequirement(20000000), 0.60);
```
Comment (Aegyo) [inline src/scripts/achievements/AchievementHandler.ts:211]: Why 20 and not 10? I don't mind, but all the other currency achievements above this start with a 1, this one sticks out.

File: src/scripts/achievements/AchievementHandler.ts [hunk 3]
```diff
@@ -210,11 +239,11 @@ class AchievementHandler {
...
+        AchievementHandler.addAchievement('I\'ve got my hands full with all of these', 'Level 3 Oak Items to the maximum level', new MaxLevelOakItemRequirement(3), 0.1, GameConstants.ExtraAchievementCategories.global, () => !challenges.list.disableOakItems.active());
+        AchievementHandler.addAchievement('Professor Oak is the best!', 'Level 8 Oak Items to the maximum level', new MaxLevelOakItemRequirement(8), 0.14, GameConstants.ExtraAchievementCategories.global, () => !challenges.list.disableOakItems.active());
+        AchievementHandler.addAchievement('Prepared for anything!', 'Level 11 Oak Items to the maximum level', new MaxLevelOakItemRequirement(11), 0.18, GameConstants.ExtraAchievementCategories.global, () => !challenges.list.disableOakItems.active() && challenges.list.disableClickAttack.active());
+        AchievementHandler.addAchievement('Prepared for anything!', 'Level 12 Oak Items to the maximum level', new MaxLevelOakItemRequirement(12), 0.18, GameConstants.ExtraAchievementCategories.global, () => !challenges.list.disableOakItems.active() && !challenges.list.disableClickAttack.active());
```
Comment (Aegyo) [inline src/scripts/achievements/AchievementHandler.ts:246]: Both called 'Prepared for anything!'

File: src/scripts/achievements/AchievementHandler.ts [hunk 4]
```diff
@@ -241,13 +272,18 @@ class AchievementHandler {
         AchievementHandler.addAchievement('Where Did All The Shiny Pokémon Go?', 'Capture 50 unique Shinies', new ShinyPokemonRequirement(50), 0.30);
         AchievementHandler.addAchievement('Nvm, found some more', 'Capture 75 unique Shinies', new ShinyPokemonRequirement(75), 0.45);
         AchievementHandler.addAchievement('Just keep swimming just keep swimming -- oooo shiny!', 'Capture 100 unique Shinies', new ShinyPokemonRequirement(100), 0.60);
-        AchievementHandler.addAchievement('I don\'t know if I can handle the next batch of shinies.', 'Capture 151 unique Shinies!', new ShinyPokemonRequirement(151), 1.50);
+        AchievementHandler.addAchievement('I don\'t know if I can handle the next batch of shinies', 'Capture 151 unique Shinies!', new ShinyPokemonRequirement(151), 1.00);
+        AchievementHandler.addAchievement('Will need sunglasses', 'Capture 250 unique Shinies!', new ShinyPokemonRequirement(250), 1.20);
+        AchievementHandler.addAchievement('What was the odds again?', 'Capture 500 unique Shinies!', new ShinyPokemonRequirement(500), 1.30);
```
Comment (Aegyo) [inline src/scripts/achievements/AchievementHandler.ts:277]:
```suggestion
        AchievementHandler.addAchievement('What were the odds again?', 'Capture 500 unique Shinies!', new ShinyPokemonRequirement(500), 1.30);
```

File: src/scripts/achievements/AchievementHandler.ts [hunk 5]
```diff
@@ -311,28 +351,65 @@ class AchievementHandler {
...
+        AchievementHandler.addAchievement('Where is nursery Joy?', 'Infect 250 Pokémon with Pokerus', new PokerusStatusRequirement(250, GameConstants.Pokerus.Infected), 0.4);
+        AchievementHandler.addAchievement('Infecting the whole world', 'Infect 1000 Pokémon with Pokerus', new PokerusStatusRequirement(1000, GameConstants.Pokerus.Infected), 0.5);
+        AchievementHandler.addAchievement('Curing them one at a time', 'Have 10 Resistant Pokémon', new PokerusStatusRequirement(10, GameConstants.Pokerus.Resistant), 0.3);
+        AchievementHandler.addAchievement('Pokémon center regular', 'Have 50 Resistant Pokémon', new PokerusStatusRequirement(50, GameConstants.Pokerus.Resistant), 0.5);
+        AchievementHandler.addAchievement('Doctor in training', 'Have 250 Resistant Pokémon', new PokerusStatusRequirement(250, GameConstants.Pokerus.Resistant), 1);
+        AchievementHandler.addAchievement('I should open my own Pokémon Center', 'Have 500 Resistant Pokémon', new PokerusStatusRequirement(500, GameConstants.Pokerus.Resistant), 1.5);
+
```
Comment (Aegyo) [inline src/scripts/achievements/AchievementHandler.ts:377]: 'Pokerus' -> 'Pokérus', and 'Have x Resistant Pokemon' -> 'Have x Pokemon Resistant to Pokerus'

```suggestion
        AchievementHandler.addAchievement('I hope it\'s not dangerous', 'Infect 20 Pokémon with Pokérus', new PokerusStatusRequirement(20, GameConstants.Pokerus.Infected), 0.1);
        AchievementHandler.addAchievement('Anyway to get rid of it?', 'Infect 50 Pokémon with Pokérus', new PokerusStatusRequirement(50, GameConstants.Pokerus.Infected), 0.2);
        AchievementHandler.addAchievement('It is the flu season', 'Infect 100 Pokémon with Pokérus', new PokerusStatusRequirement(100, GameConstants.Pokerus.Infected), 0.3);
        AchievementHandler.addAchievement('Where is nursery Joy?', 'Infect 250 Pokémon with Pokérus', new PokerusStatusRequirement(250, GameConstants.Pokerus.Infected), 0.4);
        AchievementHandler.addAchievement('Infecting the whole world', 'Infect 1000 Pokémon with Pokérus', new PokerusStatusRequirement(1000, GameConstants.Pokerus.Infected), 0.5);
        AchievementHandler.addAchievement('Curing them one at a time', 'Have 10 Pokémon Resistant to Pokérus', new PokerusStatusRequirement(10, GameConstants.Pokerus.Resistant), 0.3);
        AchievementHandler.addAchievement('Pokémon center regular', 'Have 50 Pokémon Resistant to Pokérus', new PokerusStatusRequirement(50, GameConstants.Pokerus.Resistant), 0.5);
        AchievementHandler.addAchievement('Doctor in training', 'Have 250 Pokémon Resistant to Pokérus', new PokerusStatusRequirement(250, GameConstants.Pokerus.Resistant), 1);
        AchievementHandler.addAchievement('I should open my own Pokémon Center', 'Have 500 Pokémon Resistant to Pokérus', new PokerusStatusRequirement(500, GameConstants.Pokerus.Resistant), 1.5);

```

**Review by Aegyo — CHANGES_REQUESTED** [review]
Just some text changes, and a question about one of the numbers.
Looks like it all works fine otherwise :+1:

Also with this, max achievement % is going to be.. 1050%? or something like that. Do we want to give global 250% so we get a multiple of 100?

File: src/modules/requirements/HatcheryHelperRequirement.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    constructor(helpersUnlocked: number, private bonusRequired: number) {
+        super(helpersUnlocked, GameConstants.AchievementOption.more, GameConstants.AchievementType.Hatch);
+    }
+
+    public getProgress() {
+        return Math.min(App.game.breeding.hatcheryHelpers.available().filter((h) => h.hatchBonus() >= this.bonusRequired).length, this.requiredValue);
+    }
+
+    public hint(): string {
+        return `${this.requiredValue} Hatchery helpers needs atleast ${this.bonusRequired} bonus.`;
```
[the remark this answers is not in the document]
Comment (CorgiOnNeptune) [reply src/modules/requirements/HatcheryHelperRequirement.ts:14]: Should also just be ‘need’ if hatchery helpers is plural.

File: src/scripts/achievements/AchievementHandler.ts (hunk 3 above)
Comment (PixLSteam) [reply src/scripts/achievements/AchievementHandler.ts:246]: This is intended - one is with the no click challenge, the other one is without.

**Comment by Jaaslet** [comment]
> Also with this, max achievement % is going to be.. 1050%? or something like that. Do we want to give global 250% so we get a multiple of 100?

I think 250% is too much. And i think we are going to give Hisui 50% too, when we release that one, so we will end up with a nicer number

File: src/scripts/achievements/AchievementHandler.ts (hunk 2 above)
Comment (Jaaslet) [reply src/scripts/achievements/AchievementHandler.ts:211]: 10 mil felt too easy and 50 mil felt too hard. So i ended up with this weird number

**Review by Aegyo — APPROVED** [review]

**Pull request merged**